crates/ruff_linter/src/rules/flake8_pytest_style/rules/test_functions.rs

```diff
@@ -0,0 +1,82 @@
+use crate::checkers::ast::Checker;
+use crate::rules::flake8_pytest_style::rules::helpers::is_likely_pytest_test;
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{ParameterWithDefault, StmtFunctionDef};
+use ruff_text_size::Ranged;
+
+/// ## What it does
+/// Checks for parameters of test functions with default arguments.
+///
+/// ## Why is this bad?
+/// Such a parameter will always have the default value during the test
+/// regardless of whether a fixture with the same name is defined.
+///
+/// ## Example
+///
+/// ```python
+/// def test_foo(a=1): ...
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// def test_foo(a): ...
+/// ```
+///
+/// ## Fix safety
+/// This rule's fix is marked as unsafe, as modifying a function signature can
+/// change the behavior of the code.
+///
+/// ## References
+/// - [Original Pytest issue](https://github.com/pytest-dev/pytest/issues/12693)
+#[derive(ViolationMetadata)]
+pub(crate) struct PytestParameterWithDefaultArgument {
+    parameter_name: String,
+}
+
+impl AlwaysFixableViolation for PytestParameterWithDefaultArgument {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!(
+            "Test function parameter `{}` has default argument",
+            self.parameter_name
+        )
+    }
+
+    fn fix_title(&self) -> String {
+        "Remove default argument".to_string()
+    }
+}
+
+/// PT028
+pub(crate) fn parameter_with_default_argument(
+    checker: &mut Checker,
+    function_def: &StmtFunctionDef,
+) {
+    if !is_likely_pytest_test(function_def, checker) {
+        return;
+    }
+
+    let parameters = function_def.parameters.as_ref();
+
+    for ParameterWithDefault {
+        parameter,
+        default,
+        range: pwd_range,
+    } in parameters.iter_non_variadic_params()
+    {
+        let Some(default) = default else {
+            continue;
+        };
+
+        let parameter_name = parameter.name.to_string();
+        let kind = PytestParameterWithDefaultArgument { parameter_name };
+        let diagnostic = Diagnostic::new(kind, default.range());
+
+        let edit = Edit::deletion(parameter.end(), pwd_range.end());
+        let fix = Fix::unsafe_edit(edit);
```

Comment:
I'm not sure if we should provide a fix here because it is likely to break the test:

* I'd have to test this but isn't the pytest call going to fail without the default value? 
* Assuming the call succeeds, `a` now no longer has the value `1` and instead some other default value (`None`?) This can lead to all sort of failures when running the test.

Reply:
We could make it `display-only` but I don't think we're displaying `display-only` fixes today, so we could as well just remove it. But I'm fine with either (display only or removing it)